sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1464,17 +1464,35 @@ case class ArrayContains(left: Expression, right: Expression)
     nullSafeCodeGen(ctx, ev, (arr, value) => {
       val i = ctx.freshName("i")
       val getValue = CodeGenerator.getValue(arr, right.dataType, i)
-      s"""
-      for (int $i = 0; $i < $arr.numElements(); $i ++) {
-        if ($arr.isNullAt($i)) {
-          ${ev.isNull} = true;
-        } else if (${ctx.genEqual(right.dataType, value, getValue)}) {
-          ${ev.isNull} = false;
-          ${ev.value} = true;
-          break;
+      def checkAndSetIsNullCode(body: String) = {
+        if (nullable) {
+          s"""
+             |if ($arr.isNullAt($i)) {
```

Comment:
How about the case `right.nullable = true` and 
`left.nullable = false AND left.dataType.asInstanceOf[ArrayType].containsNull = false`?

Reply:
@maropu This should be rejected in checkInputDataTypes(), no ?